reference-apps/aaos-unbundled/mediacompose/src/main/java/com/android/designcompose/reference/mediacompose/Modules.kt

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.designcompose.reference.mediacompose
+
+import android.graphics.Bitmap
+import androidx.compose.runtime.Composable
+import com.android.designcompose.ComponentReplacementContext
+import com.android.designcompose.ImageReplacementContext
+import com.android.designcompose.ListContent
+import com.android.designcompose.MeterFunction
+import com.android.designcompose.TapCallback
+import com.android.designcompose.annotation.DesignContentTypes
+import com.android.designcompose.annotation.DesignModule
+import com.android.designcompose.annotation.DesignModuleClass
+import com.android.designcompose.annotation.DesignPreviewContent
+import com.android.designcompose.annotation.DesignProperty
+import com.android.designcompose.annotation.DesignVariantProperty
+import com.android.designcompose.annotation.PreviewNode
+
+@DesignModuleClass
+class NowPlayingModule(
+    @DesignProperty(node = "#media/now-playing/title") val title: String,
+    @DesignProperty(node = "#media/now-playing/subtitle") val artist: String,
+    @DesignProperty(node = "#media/now-playing/album") val album: String,
+    @DesignProperty(node = "#media/now-playing/artwork")
+    val albumArt: @Composable (ImageReplacementContext) -> Bitmap?,
+    @DesignProperty(node = "#media/now-playing/source-icon") val appIcon: Bitmap?,
+    @DesignProperty(node = "#media/now-playing/source-name") val appName: String,
+    @DesignModule val nowPlayingProgressBarModule: NowPlayingProgressBarModule,
```

Comment:
Since this is a nested module, you need to use `@DesignModuleProperty`. It's annoying that KSP doesn't complain that `@DesignModule` is applied to the wrong type, since it is an annotation that targets function parameters, not class properties. I'm hoping KSP fixes the issue where I need to use a different annotation for parameters/properties.

Reply:
Done.